src/quill.mention.js

```diff
@@ -160,11 +163,28 @@ class Mention {
     return this.allowedChars.test(s);
   }
 
-  setMentionListPosition(startIndex) {
+  setMentionListPosition(atSignIndex) {
     const containerPos = this.quill.container.getBoundingClientRect();
-    const indexPos = this.quill.getBounds(startIndex);
-    this.mentionContainer.style.top = `${window.scrollY + containerPos.top + indexPos.bottom}px`;
-    this.mentionContainer.style.left = `${window.scrollX + containerPos.left + indexPos.left}px`;
+    const atSignBounds = this.quill.getBounds(atSignIndex);
+
+    if ((atSignBounds.left + 230) > this.quill.container.offsetWidth) {
+      this.mentionContainer.style.left = "auto";
+      this.mentionContainer.style.right = 0;
```

Comment:
You are setting the right style = 0 here but not resetting the value in the else statement. (It will always be 0)